### PR TITLE
Added Haskell language and 2 Haskell frameworks

### PR DESCRIPTION
**Pull request by toastal**
Can test @

Yesod - http://www.yesodweb.com/
Snap - http://snapframework.com/

### CODE REVIEW

File: share/apps.json [hunk 1]
```diff
@@ -1531,6 +1531,11 @@
 			"headers": { "Server": "Happy ICS Server(?:/([\\d.]+))?\\;version:\\1" },
 			"implies": "OmniTouch 8660 My Teamwork"
 		},
+		"Haskell": {
+			"website": "wiki.haskell.org/Haskell",
+			"cats": [ 27 ],
+			"headers": { "Server": "^(?:Snap|Warp)" }
```
Comment (ElbertF) [inline share/apps.json]: This line can be removed, it's covered by the Snap and Yesod entries.

File: share/apps.json [hunk 2]
```diff
@@ -3380,6 +3385,12 @@
 			"cats": [ 1 ],
 			"meta": { "generator": "sNews" }
 		},
+		"Snap": {
+			"website": "snapframework.com",
+			"cats": [ 18, 22 ],
+			"headers": { "Server": "Snap/\\d+(\\.\\d+)+" },
```
Comment (ElbertF) [inline share/apps.json]: To capture the version number this should probably be `Snap/(\\d+(?:\\.\\d+)+)\\;version:\\1`. Same for Yesod.

File: share/apps.json (hunk 1 above)
Comment (toastal) [inline share/apps.json]: Ah, I'm not entirely sure how the system works—I just looked at how the Ruby entry is set.

Comment (ElbertF) [inline share/apps.json]: No problem. Refer the wiki if you're interested:

https://github.com/ElbertF/Wappalyzer/wiki/Specification

Comment (toastal) [inline share/apps.json]: Well there it is. :) I might add a few more languages and libraries in the future then.

File: share/apps.json (hunk 2 above)
Comment (toastal) [inline share/apps.json]: Fixed!

**Comment by toastal** [comment]
This should be good to go now.

**Comment by ElbertF** [comment]
Yep, but GitHub can't merge the pull request for some reason. I'll merge it manually when I have the time.

**Comment by ElbertF** [comment]
Merged manually, thanks!

**Pull request closed without merging**